### PR TITLE
Implement QuickPoll

### PR DESCRIPTION
**Pull request by dinohamzic**

### CODE REVIEW

File: src/QVClient.ts [hunk 1]
```diff
@@ -179,6 +183,27 @@ export class QVClient {
     return this._http.postAsync(uri, model);
   }
 
+  public GetMode(model: IQVModel): Mode {
```
Comment (VoterScience) [inline src/QVClient.ts:190]: Yes!!! This is perfect.  Much easier to follow the logic.

File: src/QVClient.ts [hunk 2]
```diff
@@ -197,4 +222,28 @@ export class QVClient {
...
+    let plainUri = `/election/${shortId}/ajaxquickpoll?msg=${message}`;
+    const uri = new XC.UrlBuilder(plainUri);
+    return this._http.postAsync(uri, {});
+  }
+
+  public PostCloseQuickPoll(): Promise<void> {
+    var shortId = this._sheetId.substr(3);
+    let plainUri = `/election/${shortId}/ajaxquickpoll/done`;
+    const uri = new XC.UrlBuilder(plainUri);
+    return this._http.postAsync(uri, {});
```
Comment (VoterScience) [inline src/QVClient.ts:237]: This response from this request gives you the poll results.

File: src/QVClient.ts [hunk 3]
```diff
@@ -197,4 +222,28 @@ export class QVClient {
...
+  }
+
+  public PostSubmitResults(body: {
+    round: number;
+    results: ITallyResultsEntry[];
+  }): Promise<void> {
+    var shortId = this._sheetId.substr(3);
```
Comment (VoterScience) [inline src/QVClient.ts]: nit: can we factor this expression to its own function, like: 
 this.GetShortId();

File: src/QVClient.ts [hunk 4]
```diff
@@ -157,6 +153,14 @@ export interface IQVModel extends IQVModelEdit {
   credentialMetadata?: ICredentialMetadata;
 }
 
+export enum Mode {
+  Begin,
```
Comment (VoterScience) [inline src/QVClient.ts:157]: Add one more for "Not Yet Opened".

File: src/index.tsx [hunk 5]
```diff
@@ -853,30 +852,18 @@ export class App extends React.Component<IProps, IState> {
...
-                    ""
-                  )}/manage`}
-                  target="_blank"
-                  style={{ textAlign: "right" }}
-                >
-                  Open in legacy plugin
-                </LegacyUrl>
-              </Grid>
+              <div>
+                {this.state.Model.stage === -1 && (
```
Comment (VoterScience) [inline src/index.tsx]: Can these use the new Mode enum? Will centralize the logic.

File: src/QVClient.ts (hunk 4 above)
Comment (dinohamzic) [reply src/QVClient.ts:157]: Should `NotYetOpened` be returned when `modes.stage` is equal to `-1`? 

If that's the case, isn't that the same as `Mode.Begin`? Should I remove `Mode.Begin` and replace it with `Mode. NotYetOpened`?

**Review by VoterScience — APPROVED** [review]

**Pull request merged**